src/components/search-note.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class SearchNote extends React.Component {
+  static propTypes = {
+    onHandleTyping: PropTypes.func.isRequired
+  };
+
+  state = {
+    searchText: ""
+  };
+
+  handleTyping = (e) => {
+    const searchText = e.target.value;
+    this.setState({ searchText })
```

Comment:
Follow "single source of truth". 
`searchText` state should be store at one place. In your code, `searchText` was place in both container and it self component. Should move this state to parent.